### DISCUSSION TITLE
Android application floating windows and barrier-free controls

### DISCUSSION THREAD

**Discussion by boysj** [beeware/briefcase#1833] [Q&A]
### What is the problem or limitation you are having?

Can Android application floating windows and barrier-free controls, such as automatic clicking on specific coordinates, be implemented? If there is a corresponding document, please provide the address.

### Describe the solution you'd like

I want to know how to implement it. If it has been implemented, please tell me the document address.

### Describe alternatives you've considered

no

**Comment by freakboy3742** [comment]
Converted to a discussion because this is a generic request, rather than a specific enhancement proposal.

The short version is that no - we don't currently support Android floating windows or barrier-free controls. 

This is, at least in part, because the features you've described are Android specific APIs. In order to be added to Toga as a cross-platform API, we'd need to find a cross-platform representation of the underlying ideas. That might be possible, but it needs some design work first.

However - any platform native API is also accessible, as long as you're comfortable with your app not being cross-platform. So, if you want to convert a window to be floating, you can use the native Android APIs for enabling that feature on the Toga app. Every Toga widget and window has a `_impl` attribute, and those `_impl` objects have at least a `native` attribute, and often other native instances. Those native instances are the underlying raw Java objects that can be manipulated as if they were actual Java, using the [Chaquopy](https://chaquo.com/chaquopy/doc/current/python.html) bridging layer.
